syntax = "proto3";

package information_superhighway;

// Service definition
service InformationSuperhighwayService {
    // AI analysis request
    rpc ImageAiAnalysisRequest (ImageAnalysisRequest) returns (stream StatusReply) {}
}

// AI analysis request message
message ImageAnalysisRequest {
    int64 photo_id = 1;
    string b64image = 2;
    string model_name = 3;
}

// A basic status response message
message StatusReply {
    string message = 1;
}
